cli/azd/pkg/auth/token.go

```diff
@@ -18,61 +18,94 @@ var jwtClaimsRegex = regexp.MustCompile(`^[a-zA-Z0-9-_]*\.([a-zA-Z0-9-_]*)\.[a-z
 
 // cspell: enable
 
-func GetTenantIdFromToken(token string) (string, error) {
-	return getTidClaimFromAccessToken(token)
+// TokenClaims contains claims about a user from an access token.
+// https://learn.microsoft.com/en-us/entra/identity-platform/access-token-claims-reference.
+type TokenClaims struct {
+	PreferredUsername string `json:"preferred_username,omitempty"`
+	UniqueName        string `json:"unique_name,omitempty"`
+	GivenName         string `json:"given_name,omitempty"`
+	FamilyName        string `json:"family_name,omitempty"`
+	MiddleName        string `json:"middle_name,omitempty"`
+	Name              string `json:"name,omitempty"`
+	Oid               string `json:"oid,omitempty"`
+	TenantId          string `json:"tid,omitempty"`
+	Subject           string `json:"sub,omitempty"`
+	Upn               string `json:"upn,omitempty"`
+	Email             string `json:"email,omitempty"`
+	AlternativeId     string `json:"alternative_id,omitempty"`
+	Issuer            string `json:"iss,omitempty"`
+	Audience          string `json:"aud,omitempty"`
+	ExpirationTime    int64  `json:"exp,omitempty"`
+	IssuedAt          int64  `json:"iat,omitempty"`
+	NotBefore         int64  `json:"nbf,omitempty"`
 }
 
-// GetOidFromAccessToken extracts a string claim with the name "oid" from an access token.
-// Access Tokens are JWT and the middle component is a base64 encoded string of a JSON object
-// with claims.
-func GetOidFromAccessToken(token string) (string, error) {
-	matches := jwtClaimsRegex.FindStringSubmatch(token)
-	if len(matches) != 2 {
-		return "", errors.New("malformed access token")
+// Returns an ID associated with the account. This ID is suitable for local use, and not for any server authorization use.
+func (tc *TokenClaims) LocalAccountId() string {
+	if tc.Oid != "" {
```

Comment:
Just wondering - in practice when do you get a token back that doesn't have an oid claim?

Reply:
Ahh!  Your follow up PR explained it nicely:

> When oid claim is not present (personal accounts), the sub (subject) claim is used as the account ID instead.

Somewhat terrifying - but reasonable, I guess.

Reply:
I'll add a comment here to clarify.